packages/react-aria-components/src/Menu.tsx

```diff
@@ -163,22 +165,26 @@ function Menu<T extends object>(props: MenuProps<T>, ref: ForwardedRef<HTMLDivEl
 
 interface MenuInnerProps<T> {
   props: MenuProps<T>,
-  collection: ICollection<Node<object>>,
+  collection: BaseCollection<object>,
   menuRef: RefObject<HTMLDivElement | null>
 }
 
 function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInnerProps<T>) {
+  let {register, filterFn, inputValue, menuProps: autocompleteMenuProps} = useContext(InternalAutocompleteContext) || {};
+  // TODO: Since menu only has `items` and not `defaultItems`, this means the user can't have completly controlled items like in ComboBox,
+  // we always perform the filtering for them.
+  let filteredCollection = useMemo(() => filterFn ? collection.filter(filterFn) : collection, [collection, filterFn]);
```

Comment:
One difference to note, but the user can always pass their own `defaultFilterFn` that returns true always if they want the `items` they provide to be fully controlled (aka if they do their own filtering outside the Autocomplete and thus don't want our default filter fn to also be applied). It does feel a bit confusing though, but we'd have to add `defaultItems` to Menu if we want to make `items` fully controlled...

Reply:
I was gonna ask about this. Seems less common for menu, but if someone wanted to create a SearchAutocomplete-like component with RAC, they'd probably want to be able to control the filtered items from the server.

Reply:
yeah its not great :/. However, if we wanted to default the filterFn to return `true` always then static collections wouldn't have filtering by default though

Reply:
let's discuss this tomorrow